osc-server/src/main/java/org/osc/core/broker/util/network/NetworkSettingsApi.java

```diff
@@ -20,132 +20,93 @@
-            }
-
-            networkSettingsDto.setHostDefaultGateway(networkConfig.getProperty(this.GATEWAY));
-            try {
-                String[] dns = getDNSSettings();
-                networkSettingsDto.setHostDnsServer1(dns[0]);
-                networkSettingsDto.setHostDnsServer2(dns[1]);
-            } catch (IOException e) {
-                log.error("Failed to load DNS settings", e);
-            }
-
-        } else {
-            // only return IP in DHCP mode
-            networkSettingsDto.setDhcp(true);
-            String hostIpAddress = null;
-            try {
-                hostIpAddress = NetworkUtil.getHostIpAddress();
-            } catch (SocketException | UnknownHostException e) {
-                log.error("Failed to get host and/or ip address", e);
-            }
+            hostIpAddress = NetworkUtil.getHostIpAddress();
             networkSettingsDto.setHostIpAddress(hostIpAddress);
-            networkSettingsDto.setHostSubnetMask("");
-            networkSettingsDto.setHostDefaultGateway("");
-            networkSettingsDto.setHostDnsServer1("");
-            networkSettingsDto.setHostDnsServer2("");
-        }
-
+            networkSettingsDto.setHostSubnetMask(getIPv4LocalNetMask());
+            networkSettingsDto.setHostDefaultGateway(getDefaultGateway());
+            String[] dns = getDNSSettings();
+            networkSettingsDto.setHostDnsServer1(dns[0]);
+            networkSettingsDto.setHostDnsServer2(dns[1]);
+
+        } catch (SocketException | UnknownHostException e) {
+            log.error("Failed to get host and/or ip address", e);
+        } catch (IOException e) {
+            log.error("Failed to get DNS Server(s)", e);
+		}
         return networkSettingsDto;
     }
 
     private String[] getDNSSettings() throws IOException {
-        String[] dns = { "", "" };
+        String[] dns = { "", "", "" };
+        int index = 0;
 
-        int i = 0;
         try (FileReader fileReader = new FileReader(this.NETWORK_RESOLV);
              Scanner in = new Scanner(fileReader)) {
             while (in.hasNextLine()) {
                 String[] tokens = in.nextLine().split(" ");
                 String s = tokens[0];
                 if (s.startsWith("name") && tokens.length > 1) {
-                    dns[i++] = tokens[1];
-                    if (i > 2) {
+                    dns[index++] = tokens[1];
+                    if (index > 2) {
                         break;
                     }
                 }
             }
         }
 
-        return dns;
+    return dns;
     }
 
+    private String getIPv4LocalNetMask() {
+        String netMask="";
+        String[] cmd = { "/bin/sh", "-c", "ip addr show eth0 |grep inet|tr -s ' ' |cut -d' ' -f3" };
+        List<String> outlines=new ArrayList<String>();
+
+        ServerUtil.execWithLog(cmd, outlines, true);
+        for(String line:outlines) {
+            netMask=line;
```

Comment:
Should we break after the first line?

add log messages to see in case we dont see any line. Also log the outlines for debug purposes. same for below.

Reply:
Although, there are logs  in ServerUtil.execWithLog().But even though i am adding 1.